gprofiler/profilers/java.py

```diff
@@ -784,28 +769,34 @@ def start(self) -> None:
             self._saved_mlock = read_perf_event_mlock_kb()
             write_perf_event_mlock_kb(self._new_perf_event_mlock_kb)
 
+        # needs to run in init net NS
+        run_in_ns(["net"], lambda: proc_events.register_exit_callback(self._proc_exit_callback), 1)
```

Comment:
Did you figure out why?

Reply:
I suspect it's because the kernel listens only on the init net. I don't know netlink good enough to be able to prove/disprove that

Reply:
Planning to check further tho

Reply:
```
	/* 
	 * Events are reported with respect to the initial pid
	 * and user namespaces so ignore requestors from
	 * other namespaces.
	 */
	if ((current_user_ns() != &init_user_ns) ||
	    (task_active_pid_ns(current) != &init_pid_ns))
		return;

	/* Can only change if privileged. */
	if (!__netlink_ns_capable(nsp, &init_user_ns, CAP_NET_ADMIN)) {
		err = EPERM;
		goto out;
	}
```

Pfft, it silently ignores requests from non init PID / init user namespaces.

Reply:
Lol well played. I wonder why they're silent about it

Reply:
Still doesn't explain the net NS thing (which results in ECONNREFUSED). I'll continue checking

Reply:
```
static int cn_init(void)
{
	struct cn_dev *dev = &cdev;
	struct netlink_kernel_cfg cfg = {
		.groups	= CN_NETLINK_USERS + 0xf,
		.input	= cn_rx_skb,
	};

	dev->nls = netlink_kernel_create(&init_net, NETLINK_CONNECTOR, &cfg);
	if (!dev->nls)
		return -EIO;
```

I'll add a comment

Reply:
Added